### PR TITLE
Add `date` platform docs

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
Adds docs for the new date platform.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/81948
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/3861
- Link to developer documentation pull request: https://github.com/home-assistant/core/pull/81948

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/date.markdown [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ha_domain: date
+ha_quality_scale: internal
+ha_codeowners:
+  - '@home-assistant/core'
+ha_integration_type: entity
+---
+
+The Date integration is built for the controlling and monitoring of dates on devices.
+
+Date entities are created automatically by integrations that support them.
```
Comment (frenck) [inline source/_integrations/date.markdown]:
```suggestion
Date entities cannot be implemented manually, but can be provided by other integrations. If you are looking for a way to create a similar entity, please take a look at the [Date/Time helper](/integrations/input_datetime).
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

File: source/_integrations/date.markdown [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ha_domain: date
+ha_quality_scale: internal
+ha_codeowners:
+  - '@home-assistant/core'
+ha_integration_type: entity
+---
+
+The Date integration is built for the controlling and monitoring of dates on devices.
+
+Date entities cannot be implemented manually, but can be provided by other integrations. If you are looking for a way to create a similar entity, please take a look at the [Date/Time helper](/integrations/input_datetime).
```
Comment (spacegaier) [inline source/_integrations/date.markdown]:
```suggestion
Date entities cannot be created manually, but can be provided by other integrations. If you are looking for a way to create a similar entity, please take a look at the [Date/Time helper](/integrations/input_datetime).
```

I think the meaning here is that users cannot manually create those entities and have to use helpers instead. Then I'd say "create" is clearer here.

File: source/_integrations/date.markdown [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
+title: Date
+description: Instructions on how to set up date entities within Home Assistant.
+ha_category:
+  - Date
+ha_release: '2022.12'
```
Comment (frenck) [inline source/_integrations/date.markdown]:
```suggestion
ha_release: '2023.6'
```

**Review by frenck — APPROVED** [review]

**Pull request merged**